File: 프로그래머스/2/178870. 연속된 부분 수열의 합/연속된 부분 수열의 합.py
def solution(sequence, k):
    answer = []
    
    n = len(sequence)
    left, right = 0, 0
    cur_sum = sequence[0]
    min_len = float('inf')
    
    while right < n:
        if cur_sum == k:
            if right - left < min_len:
                min_len = right - left
                answer = [left, right]
        if cur_sum >= k:
            cur_sum -= sequence[left]
            left += 1
        else:
            right += 1
            if right < len(sequence):
                cur_sum += sequence[right]
            
    return answer

File: 프로그래머스/2/178870. 연속된 부분 수열의 합/연속된 부분 수열의 합.js
function solution(sequence, k) {
    var answer = [];
    let n = sequence.length
    let left = 0
    let right = 0
    let sum = sequence[0]
    let min_len = Number.MAX_SAFE_INTEGER
    
    while(right < n) {
        if(sum == k) {
            if(right - left + 1 < min_len){
                min_len = right - left + 1
                answer = [left, right]
            }
        }
        
        if(sum >= k) {
            sum -= sequence[left]
            left += 1
        }
        else {
            right += 1
            if(right < n) sum += sequence[right]
        }
    }
    return answer;
}